backupccl: add cloud backup restore tests to nightly test suite

msbutler opened the pull request:
Previously, the unit tests in ccl/backupccl/backup_cloud_test.go were not run apart of CI as they did not have access to actual external storage containers and proper credentials. This patch adds these tests to a nightly test suite build on team city agents with access to actual cloud resources.

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/90587)
<!-- Reviewable:end -->

msbutler commented:
I'd fail a bash coding interview if you gave me one, so please check if I did this properly!

stevendanna commented on build/teamcity/cockroach/nightlies/cloud_unit_tests_impl.sh:
```diff
@@ -22,29 +22,37 @@ export AWS_SHARED_CREDENTIALS_FILE="$PWD/.aws/credentials"
 export AWS_CONFIG_FILE="$PWD/.aws/config"
 log_into_aws
 
+bazel_test_env = "--test_env=GO_TEST_WRAP_TESTV=1 \
```
This should be `bazel_test_env="...` (with no spaces) otherwise it'll probably fail with something like `bazel_test_env: command not found`

msbutler replied:
Thanks!

stevendanna commented on build/teamcity/cockroach/nightlies/cloud_unit_tests_impl.sh:
```diff
@@ -22,29 +22,37 @@ export AWS_SHARED_CREDENTIALS_FILE="$PWD/.aws/credentials"
+    || exit_status=$?
+
+$BAZEL_BIN/pkg/cmd/bazci/bazci_/bazci --process_test_failures -- test --config=ci \
+    //pkg/ccl/backupccl/backupcl_test --test_filter=TestCloudBackupRestore* \
```
[nit] It doesn't matter, but the test filters are a regular expression rather than a glob. They also match any part of the word so you don't need to account for training characters. If you want to match all tests that start with TestCloudBackupRestore then you want something like `^TestCloudBackupRestore'.

msbutler replied:
aha! Didn't realize these were regex expressions!

adityamaru commented:
If you could just confirm the diff runs as expected in the nightly that'd be great. You can switch TC to just use your branch as the head and then kickoff a manual run of the cloud unit test job.

msbutler commented:
hm, the custom run failed, see [here.](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7129106?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true)

it seems that passing this `bazel_test_env` bash var into the bazel cmd causes the bazel command to improperly parse things, compared to master's [run](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7130091?showRootCauses=false&showLog=7130091_3_4950.4954.6159) currently. 

I'm going to try putting parenthesis around the `bazel_test_env` variable in the bazel cmd. hopefully that causes the variable to evaluate and expand.

If this doesn't work, I may switch my approach and put the cloud tests in their own package.

msbutler commented:
I believe that --test_filter is quite blunt and will filter on all test targets, but i'll double check.

Also, is it just me, or does it seem like a google access key is leaking into the build log in the latest failure???
https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7131087?showRootCauses=false&expandBuildProblemsSection=true&expandBuildTestsSection=true&expandBuildChangesSection=true

adityamaru commented:
We don't consider accesskey as a secret, historically we've though of it as a username. secretkey however is redacted.

msbutler committed:
backupccl: add cloud backup restore test to nightly test suite

Previously, the unit tests in ccl/backupccl/backup_cloud_test.go were not run
apart of CI as they did not have access to actual external storage containers
and proper credentials. This patch adds these tests to a nightly test suite
built on team city agents with access to actual cloud resources.

Release note: None

msbutler commented:
@adityamaru hrm, it seems that adding `test_filter` silently clobbers the other test targets https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7133512?buildTab=tests

If anyone has other bash suggestions, on this [commit](https://github.com/cockroachdb/cockroach/commit/75ea0e0f5b3cf324bbeb30ac251319ff36597b51), I'm all ears

msbutler commented:
latest cloud unit test worked! https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/7155072?buildTab=tests&pager.currentPage=5

thanks @stevendanna for the debugging help and bash wisdom. turns out i needed to pass the var as an array. This is a very helpful tool to use on within goland https://github.com/koalaman/shellcheck

adityamaru commented:
Nice! LGTM, we have a couple ignored tests in that green build. Do we know why they aren't running?

<img width="1587" alt="Screen Shot 2022-10-26 at 5 26 23 PM" src="https://user-images.githubusercontent.com/13837382/198141624-3c2985ec-2771-481a-a0be-c64a34a177d5.png">

stevendanna commented on build/teamcity/cockroach/nightlies/cloud_unit_tests_impl.sh:
```diff
@@ -22,29 +22,37 @@ export AWS_SHARED_CREDENTIALS_FILE="$PWD/.aws/credentials"
 export AWS_CONFIG_FILE="$PWD/.aws/config"
 log_into_aws
 
+bazel_test_env=(--test_env=GO_TEST_WRAP_TESTV=1 \
```
Just an FYI, no need to change:  I'm pretty sure you don't need the `\` here anymore since you've changed it to an array.

stevendanna reviewed (approved):
LGTM! I like that you went the array route. It does really clean things up once you know about them.
